server/src/test/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldTypeTests.java

```diff
@@ -99,6 +136,36 @@ public void testCreateKnnQuery() {
         DenseVectorFieldType cosineField = new DenseVectorFieldType(
             "f",
             Version.CURRENT,
+            DenseVectorFieldMapper.ElementType.FLOAT32,
+            3,
+            true,
+            VectorSimilarity.cosine,
+            Collections.emptyMap()
+        );
+        e = expectThrows(IllegalArgumentException.class, () -> cosineField.createKnnQuery(new float[] { 0.0f, 0.0f, 0.0f }, 10, null));
+        assertThat(e.getMessage(), containsString("The [cosine] similarity does not support vectors with zero magnitude."));
+    }
+
+    public void testByteCreateKnnQuery() {
+        DenseVectorFieldType unindexedField = new DenseVectorFieldType(
+            "f",
+            Version.CURRENT,
+            DenseVectorFieldMapper.ElementType.INT8,
+            3,
+            false,
+            VectorSimilarity.cosine,
+            Collections.emptyMap()
+        );
+        IllegalArgumentException e = expectThrows(
+            IllegalArgumentException.class,
+            () -> unindexedField.createKnnQuery(new float[] { 0.3f, 0.1f, 1.0f }, 10, null)
+        );
+        assertThat(e.getMessage(), containsString("to perform knn search on field [f], its mapping must have [index] set to [true]"));
+
+        DenseVectorFieldType cosineField = new DenseVectorFieldType(
+            "f",
+            Version.CURRENT,
+            DenseVectorFieldMapper.ElementType.FLOAT32,
```

Comment:
Should this be `DenseVectorFieldMapper.ElementType.INT8` ?

Reply:
Good catch! Thank you. Copy and paste error.